server/graphql/v2/mutation/TagMutations.ts

```diff
@@ -0,0 +1,70 @@
+import type { Request } from 'express';
+import { GraphQLList, GraphQLNonNull, GraphQLObjectType, GraphQLString } from 'graphql';
+
+import models from '../../../models';
+import { canEditExpenseTags } from '../../common/expenses';
+import { NotFound, Unauthorized } from '../../errors';
+import { ExpenseReferenceInput, fetchExpenseWithReference } from '../input/ExpenseReferenceInput';
+import { fetchOrderWithReference, OrderReferenceInput } from '../input/OrderReferenceInput';
+import { Expense } from '../object/Expense';
+import { Order } from '../object/Order';
+
+const TagResponse = new GraphQLObjectType({
+  name: 'TagResponse',
+  fields: () => ({
+    order: {
+      type: Order,
+    },
+    expense: {
+      type: Expense,
+    },
+  }),
+});
+
+const tagMutations = {
+  setTags: {
+    type: new GraphQLNonNull(TagResponse),
+    args: {
+      tags: {
+        type: new GraphQLList(GraphQLString),
+        description: 'Tags associated with the object being updated',
+      },
+      order: {
+        type: OrderReferenceInput,
+      },
+      expense: {
+        type: ExpenseReferenceInput,
+      },
+    },
+    resolve: async (_: void, args, req: Request) => {
+      if (!req.remoteUser) {
+        throw new Unauthorized();
+      }
+
+      if (args.order) {
```

Comment:
To have a clear pattern in case they provide multiple entities

```suggestion
      if ([args.order, args.expense].filter(Boolean).length !== 1) {
        throw new Error('A single order or expense must be provided');
      }

      if (args.order) {
```